Code/GraphMol/ChemReactions/testReaction.cpp

```diff
@@ -7390,6 +7390,24 @@ void testDblBondCrash() {
   }
 }
 
+void testGithub3097() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdInfoLog) << "Testing Github issue # 3097" << std::endl;
+
+  std::string smi =
+      "[c:6][n:7][c:8].[N:14]#[N:15]>>[C:6].[C:8][N:7]=[N+:14]=[N-:15]";
+  ChemicalReaction *rxn = RxnSmartsToChemicalReaction(smi);
```

Comment:
Oh, please make sure `rxn` does not leak:

```suggestion
  std::unique_ptr<ChemicalReaction> rxn{RxnSmartsToChemicalReaction(reaction)};
```